### PR TITLE
Trim executor options based on included options

### PR DESCRIPTION
**Pull request by sabeegrewal**
Trimming options based on included options (not excluded options). When new params are added, they'll need to be explicitly added to the `include` list in order to be passed to executors. This ensures we don't accidentally pass unneeded options to executors by mistake. 👍

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala [hunk 1]
```diff
@@ -389,14 +389,15 @@ object EventHubsConf extends Logging {
...
    */
   private[spark] def trim(ehConf: EventHubsConf): EventHubsConf = {
+    // These are the options needed by Spark executors
+    val include = Seq("eventhubs.connectionString",
+                      "eventhubs.consumerGroup",
+                      "eventhubs.receiverTimeout",
+                      "eventhubs.operationTimeout",
+                      "useSimulatedClient")
+
+    ehConf.settings.asScala.filter(tuple => include.contains(tuple._1))
```
Comment (SreeramGarlapati) [inline core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala]: u intend to return the filtered list right? shouldn't this be the last line of this method as opposed to `ehConf` ?

```
ehConf.settings.asScala.filter(tuple => include.contains(tuple._1))
```

Comment (sabeegrewal) [reply core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala]: No, we're returning the `EventHubsConf` which holds the internal list. So, we filter the list, and then return the ehConf which wraps it.

Comment (SreeramGarlapati) [reply core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala]: do you mean `filter` modifies the original collection - i doubt it!!
It should return a new settings map with the filtered collection - after running your lamda.

if the trim method is called infrequently & not called in hot path - something like - for every batch - having it to return a copy of settings should be fine too.

Comment (sabeegrewal) [reply core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala]: Ah shoot, `retain` modifies the original collection. I'll update it...

**Pull request merged**